magma_ff/create_metawfr.py

```diff
@@ -558,6 +562,42 @@ def cast_parameter_value(self, parameter_value):
         return result
 
 
+def get_files_for_file_formats(file_items, file_formats, requirements=None):
+    """Gather file UUIDs for File items matching the file format
+    and meeting the requirements.
+
+    :param file_items: The files to sort through
+    :type file_items: list(dict)
+    :param file_formats: Accepted formats of files to get
+    :type file_formats: list or set
+    :param requirements: Requirements a file must meet in order to
+        be acceptable, as key, value pairs of property names, lists
+        of acceptable property values
+    :type requirements: dict
+    :returns: File UUIDs of files meeting file format and
+        requirements
+    :rtype: list(str)
+    :raises MetaWorkflowRunCreationError: If no files found to meet
+        file format/other requirements
+    """
+    result = []
+    for file_item in file_items:
+        requirements_met = True
+        if requirements:
+            for key, accepted_values in requirements.items():
+                key_value = file_item.get(key)
+                if key_value not in accepted_values:
+                    requirements_met = False
+                    break
+        if requirements_met is False:
+            continue
+        file_item_format = file_item.get(FILE_FORMAT, {}).get(FILE_FORMAT)
```

Comment:
I am not sure what this line does, you can get an error if the first get does return {}

Reply:
it would be helpful to have a comment that explains why we set an empty dictionary